### PR TITLE
Fix do_update_contribution and use `VaultPhaseChanged` event

### PR DESCRIPTION
**Pull request by rjman-ljm**

### CODE REVIEW

File: pallets/crowdloans/src/lib.rs [hunk 1]
```diff
@@ -153,44 +153,38 @@ pub mod pallet {
     #[pallet::generate_deposit(pub(super) fn deposit_event)]
     pub enum Event<T: Config> {
         /// New vault was created
-        /// [para_id, ctoken_id]
-        VaultCreated(ParaId, AssetIdOf<T>),
+        /// [para_id, vault_id, ctoken_id, phase, contribution_strategy, cap, end_block, trie_index]
+        VaultCreated(
```
Comment (rjman-ljm) [inline pallets/crowdloans/src/lib.rs:157]: crowdloans-subquery is based on event to record Vaults, we need more detailed vault information there.

File: pallets/crowdloans/src/lib.rs [hunk 2]
```diff
@@ -153,44 +153,38 @@ pub mod pallet {
...
+            ParaId,
+            u32,
```
Comment (GopherJ) [inline pallets/crowdloans/src/lib.rs:159]: I'm thinking that we can define two alias to make things more clear. 

```
pub type TrieIndex = u32;
pub type VaultId = u32;
```

we can put them inside primitives

File: pallets/crowdloans/src/lib.rs [hunk 3]
```diff
@@ -153,44 +153,38 @@ pub mod pallet {
...
-        /// [para_id]
-        VaultExpired(ParaId),
+        /// [para_id, vault_id, pre_phase, now_phase]
+        VaultPhaseChanged(ParaId, u32, VaultPhase, VaultPhase),
         /// Vault is trying to do contributing
         /// [para_id, contributor, amount, referral_code]
         VaultDoContributing(ParaId, T::AccountId, BalanceOf<T>, Vec<u8>),
         /// Vault is trying to do withdrawing
         /// [para_id, amount, target_phase]
         VaultDoWithdrawing(ParaId, BalanceOf<T>, VaultPhase),
```
Comment (GopherJ) [inline pallets/crowdloans/src/lib.rs:184]: we can also add vault id to make things more clear although it's not useful for XCM

**Comment by GopherJ** [comment]
LGTM now. Great job @rjman-ljm !

**Pull request merged**